### PR TITLE
Add `directoryMode` option

### LINKED ISSUE #13: Add `directoryMode` option
The module functions offer no way to specify mode for implicitly created directories so they always create directories world writable. If a user wants to ensure secure directories they have to avoid these methods. Accepting a directory mode would make it possible to use them securely.

World writable resources such as config files can allow other users to control program behavior. In some cases there's code injection through the config file, which can lead to privilege elevation. World writable directories have the same weakness since they allow adding files within them. A world writable config directory `/etc/froznator/conf.d` offers a route to controlling program behavior through a new config file. On Linux the umask should usually prevent this but it's not guaranteed. [Secure coding standards](https://wiki.sei.cmu.edu/confluence/display/c/FIO06-C.+Create+files+with+appropriate+access+permissions) recommend always specifying a restricted mode so files and directories are secure in any environment. MITRE calls this [CWE-732](https://cwe.mitre.org/data/definitions/732.html), one of the [top 25 most dangerous](https://cwe.mitre.org/top25/archive/2020/2020_cwe_top25.html) vulnerabilities.

These lines always use `mkdir` `mkdirSync` without a mode. If you used `move-file` to eg install default config files, you could end up with a world writable config directory.
https://github.com/sindresorhus/move-file/blob/bbabd081116d02363b95d01c82bcb9d9dea40e2a/index.js#L22
https://github.com/sindresorhus/move-file/blob/bbabd081116d02363b95d01c82bcb9d9dea40e2a/index.js#L50

I have reports for a couple of your other projects but I'll wait for your thoughts before peppering you with them.

This came out of research with CodeQL.

### PR DESCRIPTION
**Pull request by dittyroma**
Adds an option `directoryMode` which enables controlling permissions of implicitly created directories.

Closes #13.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -59,6 +59,13 @@ Default: `true`
 
 Overwrite existing destination file.
 
+##### directoryMode
+
+Type: `number` | `string`\
```
Comment (sindresorhus) [inline readme.md:64]: I would prefer to only document the number support. The string support is only in Node.js for legacy reasons.

File: readme.md [hunk 2]
```diff
@@ -59,6 +59,13 @@ Default: `true`
...
+Default: `0o777`
+
+Permissions for created directories.
```
Comment (sindresorhus) [inline readme.md:67]:
```suggestion
[Permissions](https://en.wikipedia.org/wiki/File-system_permissions#Numeric_notation) for created directories.
```

Comment (sindresorhus) [reply readme.md:67]: You also need to mention that it does nothing on Windows.

**Comment by dittyroma** [comment]
Thanks for reviewing. Updated all of that.

**Comment by sindresorhus** [comment]
Readme and index.d.ts needs to be in sync.

**Comment by dittyroma** [comment]
Made that change.

**Comment by sindresorhus** [comment]
Still not in sync.

File: index.d.ts [hunk 3]
```diff
@@ -6,6 +6,13 @@ declare namespace moveFile {
 		@default true
 		*/
 		readonly overwrite?: boolean;
+
+		/**
+		Permissions for created directories.
```
Comment (sindresorhus) [inline index.d.ts:11]:
```suggestion
		
[Permissions](https://en.wikipedia.org/wiki/File-system_permissions#Numeric_notation) for created directories.
```

**Comment by dittyroma** [comment]
Thank you @sindresorhus!

**Pull request merged**